<launch>
<include file="$(find jimmy_description)/launch/state.launch" />
  
  <node type="raspicam_node" pkg="raspicam_node" name="raspicam_node">
    <param name="camera_info_url" value="package://jimmy_drive/cfg/cal.yml"/>
    <param name="width" value="640"/>
    <param name="height" value="480"/>
    <!-- We are running at 90fps to reduce motion blur -->
    <param name="framerate" value="15"/>
  	<param name="shutterSpeed" value="0"/>
    <param name="hFlip" value="true"/>
    <param name="vFlip" value="true"/>
    <param name="exposure_mode" value="sports"/>
    <param name="brightness" value="52"/>

    <param name="camera_frame_id" value="raspicam"/> 
  </node>
  
  <node pkg="apriltags_ros" type="apriltag_detector_node" name="apriltag_detector" output="screen">
    <remap from="image_rect" to="/raspicam_node/image" />
    <remap from="camera_info" to="/raspicam_node/camera_info" />

    <param name="image_transport" type="str" value="compressed" />

    <param name="tag_family" type="str" value="36h11" />

    <param name="projected_optics" type="bool" value="true" />

    <rosparam param="tag_descriptions">[
      {id: 0, size: 0.163512},
      {id: 1, size: 0.163513},
      {id: 2, size: 0.163513}]
    </rosparam>
  </node>

  
</launch>
